src/helpers/parser.ts

```diff
@@ -49,38 +40,69 @@ export const gitLinkedIssueParser = async ({ owner, repo, pull_number }: GitPars
   }
 };
 
-export const gitLinkedPrParser = async ({ owner, repo, issue_number }: GitParser) => {
+export const gitLinkedPrParser = async ({ owner, repo, issue_number }: GitParser): Promise<HTMLElement[]> => {
```

Comment:
why does it return html elements?

Reply:
Because it's parsed html 🤔

Reply:
`gitLinkedPrParser` and `getLinkedPrs` should be one function, I think there's no need to separate

Reply:
> `gitLinkedPrParser` and `getLinkedPrs` should be one function, I think there's no need to separate

I don't understand anymore, you said you wanted `gitLinkedPrParser` generic, now it is and you want it to go back to how it was before.

I just separated it and made `gitLinkedPrParser` generic returning array of PRs and the other function handles the response and converts it to return either the latest PR or array of PR with full href ready to use

Reply:
yes `gitLinkedPrParser` should be generic in a sense that it retrieves HTML, parses it, and returns an array of PRs. It doesn't need any other functionality. Everything else should be done by functions calling `gitLinkedPrParser`.

Currently `gitLinkedPrParser` doesn't return array of PRs but just HTML elements, we don't need just HTML anywhere in the code.

Reply:
> Currently `gitLinkedPrParser` doesn't return array of PRs but just HTML elements, we don't need just HTML anywhere in the code.

Please check it out now

Reply:
looks mostly good, I'm sorry if there was miscommunication between us

Reply:
> looks mostly good, I'm sorry if there was miscommunication between us

Thanks @whilefoo, any other edits I have to make.

You used keyword *mostly*